ui/raidboss/emulator/translations.ts

```diff
@@ -0,0 +1,225 @@
+import { Lang } from '../../../resources/languages';
+import { LocaleObject, LocaleText } from '../../../types/trigger';
+
+export const defaultLocale = <T>(lang: Lang, obj: LocaleObject<T>): T => {
+  return obj[lang] ?? obj['en'];
+};
+
+export const emulatorButtons = {
+  yes: {
+    en: 'Yes',
+  },
+  no: {
+    en: 'No',
+  },
+  close: {
+    en: 'Close',
+  },
+  done: {
+    en: 'Done<span class="doneBtnTimeout"></span>',
+  },
+  loadNetworkLog: {
+    en: 'Load Network Log',
+  },
+  exportDatabase: {
+    en: 'Export DB',
+  },
+  importDatabase: {
+    en: 'Import DB',
+  },
+  clearDatabase: {
+    en: 'Clear DB',
+  },
+};
+
+export const emulatorTitle: LocaleText = {
+  en: 'Cactbot Raid Emulator',
+};
+
+export const emulatorImportModal = {
+  title: {
+    en: 'Log File Import Progress',
+  },
+  body: {
+    en: `<h3>Latest encounter:</h3>
+    Zone: <span class="zone"></span><br />
+    Encounter: <span class="encounter"></span><br />
+    Start: <span class="start"></span><br />
+    End: <span class="end"></span><br />
+    Duration: <span class="durMins"></span>m<span class="durSecs"></span>s<br />
+    Pull Duration: <span class="pullMins"></span>m<span class="pullSecs"></span>s<br />
+    Started By: <span class="startedBy"></span><br />
+    End Status: <span class="endStatus"></span><br />
+    Line Count: <span class="lineCount"></span><br />`,
+  },
+};
+
+export const emulatorDeleteModal = {
+  title: {
+    en: 'Delete Encounter Database',
+  },
+  body: {
+    en: '<p>You are about to delete the encounter database. Are you sure?</p>',
+  },
+};
+
+export const emulatorIntroModal = {
+  title: {
+    en: 'Introduction',
+  },
+  body: {
+    en: `<p>Welcome to the Raid Emulator.</p>
```

Comment:
The reason I decided to move the whole body into the translation block is because I don't want to hardcode the locations where the connected/disconnected text can go here, or the various values for the `Latest Encounter` text in the import modal.

I guess this could be constructed as an array of strings for before the connected/disconnected text, the connected/disconnected blurbs, and then array of strings for after, but that's not very nice looking either.

It also could have gone directly in the html and just hide/show the correct language blocks, but then that becomes annoying to deal with falling back to english by default.

Ultimately as long as two elements are in the block with classes `websocketConnected` and `websocketDisconnected` this should function no matter what other HTML is used, so maybe it makes sense to comment above the `body` property with the required elements, or write a test to check for them?

Reply:
The introduction text is one example of what I meant, but the `emulatorImportModal.body` is really the example that makes less sense to me.  I think that one would be better broken up into individual strings for each of the entries (and maybe minutes and seconds?) and then the ordering/spans/classes can be left in the html.

I think mostly having this baked into translation blocks creates fragility.  If we ever want to reorder them / remove one / add another, it both can be done incorrectly but also adds a lot of friction to do so.  I think it's one thing to have just a `<strong>OVERLAY_WS</strong>` where it's just style, but it's quite another to have `Duration: <span class="durMins"></span>m<span class="durSecs"></span>s<br />`.  I'm really just asking for a few more spans with class names that you can find and fill in with text.

Can you explain some more why you don't want to hardcode things in `Latest Encounter`? I'm not sure what you feel like you can't do in another way.

Reply:
The biggest issue here is the order of grammar in other languages. e.g. Japanese/Chinese may have the status before the label instead of afterwards. I guess if it's fine to enforce that ordering, it's fine for those.

The other issue here is duration, displaying `#m#s` isn't going to work for all languages, so then does every language format time strings as `#<something for minutes>#<something for seconds>` or do they vary in ordering? Do we instead need a `translateDuration` function that returns a lang-specific string for a given duration in ms?

Reply:
I think I just assumed that we could do `label: text` like we do in a bunch of other places in cactbot (e.g. config).  Google translate suggests that you could make `m` and `s` translatable in `#m#s` and it should work for Chinese and Japanese.

Reply:
Alright, I'll take another pass at this and see if I can get everything sorted.